app/models/programme_detail.rb

```diff
@@ -80,10 +97,18 @@ def age_range_valid
   end
 
   def programme_start_date_valid
-    if [day, month, year].all?(&:blank?)
+    if [start_day, start_month, start_year].all?(&:blank?)
       errors.add(:programme_start_date, :blank)
     elsif !programme_start_date.is_a?(Date)
       errors.add(:programme_start_date, :invalid)
     end
   end
+
+  def programme_end_date_valid
+    if [end_day, end_month, end_year].all?(&:blank?)
```

Comment:
Understand that you've gone with the same validations as start date, but can we confirm with @edwardhorsford that a programme end date is actually required? Could the trainees programme be ongoing still?

**If it's not required**, what should happen when an invalid date is inputted? The prototype ignores the invalid input and shows "Not provided", whereas this PR shows an error. However, if it's not required, asking people to "Please input a valid date" would be confusing.

Reply:
checked with Ed, it's required.